File: app/src/main/java/ua/com/foxminded/locationtrackera/model/auth/AuthNetwork.java
package ua.com.foxminded.locationtrackera.model.auth;

import io.reactivex.rxjava3.core.Single;
import ua.com.foxminded.locationtrackera.util.Result;

public interface AuthNetwork {

    Single<Result<Void>> firebaseRegister(String username, String email, String password);

    Single<Result<Void>> firebaseLogin(String email, String password);

    Single<Result<Void>> resetPassword(String email);

    boolean isUserLoggedIn();

    void logout();
}

File: models/src/main/java/ua/com/foxminded/locationtrackera/models/auth/AuthNetwork.kt
package ua.com.foxminded.locationtrackera.models.auth

import ua.com.foxminded.locationtrackera.models.util.Result

interface AuthNetwork {

    suspend fun firebaseRegister(username: String, email: String, password: String): Result<Void?>
    suspend fun firebaseLogin(email: String, password: String): Result<Void?>
    suspend fun resetPassword(email: String): Result<Void?>
    fun logout()
    fun isUserLoggedIn(): Boolean
}
